lib/ansible/modules/windows/win_domain_computer.ps1

```diff
@@ -0,0 +1,183 @@
+#!powershell
+#
+# Copyright: (c) 2017, AMTEGA - Xunta de Galicia
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+
+
+# ------------------------------------------------------------------------------
+$ErrorActionPreference = "Stop"
+
+# Preparing result
+$result = @{}
+Set-Attr $result "changed" $false
+
+# Parameter ingestion
+$params = Parse-Args $args -supports_check_mode $true
+
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -default $false | ConvertTo-Bool
+$diff_support = Get-AnsibleParam -obj $params -name "_ansible_diff" -default $false | ConvertTo-Bool
+
+# ------------------------------------------------------------------------------
+Function Get-DesiredState($params) {
+  $name = Get-AnsibleParam -obj $params -name "name" -failifempty $true -resultobj $result
+  $sam_account_name = Get-AnsibleParam -obj $params -name "sam_account_name" -default "$name$"
+  If (-not $sam_account_name.EndsWith("$")) {
+    Fail-Json $result "sam_account_name must end in $"
+  }
+  $enabled = Get-AnsibleParam -obj $params -name "enabled" -default $true | ConvertTo-Bool
+  $description = Get-AnsibleParam -obj $params -name "description" -default "" -resultobj $result
+  $state =(Get-AnsibleParam -obj $params -name "state" -ValidateSet "Present","Absent" -default "Present").ToLower()
```

Comment:
I don't think the `ToLower()` is necessary. Just get `Get-AnsibleParam -obj $params -name "state" -default "present" -validateset "absent", "present`

Any checks with `if ($state -eq "present")` will work regardless of the case as `-eq` is not case sensitive.